unit_test/sparse/Test_Sparse_csc2csr.hpp

```diff
@@ -0,0 +1,127 @@
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Siva Rajamanickam (srajama@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include "KokkosSparse_csc2csr.hpp"
+#include "KokkosKernels_TestUtils.hpp"
+
+namespace Test {
+template <class ScalarType, class LayoutType, class ExeSpaceType>
+void doCsc2Csr(size_t m, size_t n, ScalarType min_val, ScalarType max_val) {
+  RandCscMat<ScalarType, LayoutType, ExeSpaceType> cscMat(m, n, min_val,
+                                                          max_val);
+  constexpr int league_size = 32;
+
+  auto csrMat = KokkosSparse::csc2csr(
+      cscMat.get_m(), cscMat.get_n(), cscMat.get_nnz(), cscMat.get_vals(),
+      cscMat.get_row_ids(), cscMat.get_col_map(), league_size);
+
+  auto csc_row_ids = cscMat.get_row_ids();
+  auto csc_col_map = cscMat.get_col_map();
+  auto csc_vals    = cscMat.get_vals();
+
+  auto csr_col_ids = csrMat.graph.entries;
+  auto csr_row_map = csrMat.graph.row_map;
+  auto csr_vals    = csrMat.values;
+
+  for (int j = 0; j < cscMat.get_n(); ++j) {
+    auto col_start = csc_col_map(j);
+    auto col_len   = csc_col_map(j + 1) - col_start;
+
+    for (int k = 0; k < col_len; ++k) {
+      auto i = col_start + k;
+
+      auto row_start = csr_row_map(csc_row_ids(i));
+      auto row_len   = csr_row_map(csc_row_ids(i) + 1) - row_start;
+      auto row_end   = row_start + row_len;
+
+      if (row_len == 0) continue;
+
+      // Linear search for corresponding element in csr matrix
+      int l = row_start;
+      while (l < row_end && csr_col_ids(l) != j) {
+        ++l;
+      }
+
+      if (l == row_end)
+        FAIL() << "csr element at (i: " << csc_row_ids(i) << ", j: " << j
+               << ") not found!" << std::endl;
+
+      ASSERT_EQ(csc_vals(i), csr_vals(l))
+          << "(i: " << csc_row_ids(i) << ", j: " << j << ")" << std::endl;
+    }
+  }
+}
+
+template <class LayoutType, class ExeSpaceType>
+void doAllScalarsCsc2Csr(size_t m, size_t n, int min, int max) {
+  doCsc2Csr<float, LayoutType, ExeSpaceType>(m, n, min, max);
+  doCsc2Csr<double, LayoutType, ExeSpaceType>(m, n, min, max);
+  doCsc2Csr<Kokkos::complex<float>, LayoutType, ExeSpaceType>(m, n, min, max);
+  doCsc2Csr<Kokkos::complex<double>, LayoutType, ExeSpaceType>(m, n, min, max);
+}
+
+template <class ExeSpaceType>
+void doAllLayoutsCsc2Csr(size_t m, size_t n, int min, int max) {
+  doAllScalarsCsc2Csr<Kokkos::LayoutLeft, ExeSpaceType>(m, n, min, max);
+  doAllScalarsCsc2Csr<Kokkos::LayoutRight, ExeSpaceType>(m, n, min, max);
+}
+
+template <class ExeSpaceType>
+void doAllCsc2csr(size_t m, size_t n) {
+  int min = 1, max = 10;
+  doAllLayoutsCsc2Csr<ExeSpaceType>(m, n, min, max);
+}
+
+TEST_F(TestCategory, sparse_csc2csr) {
+  // Square cases
+  for (size_t dim = 4; dim < 1024; dim *= 4)
```

Comment:
Should we throw a wrench or two in this and test for columns that are empty or matrices that are empty in case a distributed application has not data on a given process?

Reply:
The csc matrix is randomly generated so lots of wrenches are being thrown in : )

Upon failure, the seed is printed out so that test failures are reproducible - see https://github.com/kokkos/kokkos-kernels/pull/1342/files#diff-4b273c749247dc0f15e60c360e769521bd0b7f754338dab3e6cd15f1dd900fffR517.

It's unlikely that the csc matrix would be 100% sparse, I will add a test case for that.